### PR TITLE
Refactored golang-pgx sample app

### PR DESCRIPTION
**Pull request by ericharmeling**
Follow-up on https://github.com/cockroachdb/docs/pull/10851.

This PR includes the following changes:
- Added new hello world tutorial
- Updated existing example to tutorial to fit refactor template

Related to https://github.com/cockroachlabs/hello-world-go-pgx/commit/b15a222556b9e80610973af9cb7fa14251927e0d and https://github.com/cockroachlabs/example-app-go-pgx/pull/4

After this is approved, I'll propagate to 20.2 docs, merge https://github.com/cockroachlabs/example-app-go-pgx/pull/4, and then merge this PR.

### CODE REVIEW

File: v21.1/build-a-go-app-with-cockroachdb.md [hunk 1]
```diff
@@ -13,93 +13,92 @@ referral_id: docs_hello_world_go_pgx
...
-</div>
+## Step 2. Get the code
 
-<div class="filter-content" markdown="1" data-scope="cockroachcloud">
+Clone the code's GitHub repo:
 
-{% include_cached copy-clipboard.html %}
-~~~ go
-{% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-go-pgx/cockroachcloud/main.go %}
+{% include copy-clipboard.html %}
```
Comment (ianjevans) [inline v21.1/build-a-go-app-with-cockroachdb.md:70]:
```suggestion
{% include_cached copy-clipboard.html %}
```

File: v21.1/build-a-go-app-with-cockroachdb.md [hunk 2]
```diff
@@ -13,93 +13,92 @@ referral_id: docs_hello_world_go_pgx
...
+{% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-go-pgx/master/dbinit.sql %}
 ~~~
 
-{% include {{page.version.version}}/app/cc-free-tier-params.md %}
+The `main.go` file contains the code for `INSERT`, `SELECT`, `UPDATE`, and `DELETE` SQL operations. The file also executes the `main` method of the program.
 
-</section>
+{% include_cached copy-clipboard.html %}
+~~~ go
+{% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-go-pgx/master/main.py %}
```
Comment (ianjevans) [inline v21.1/build-a-go-app-with-cockroachdb.md:94]:
```suggestion
{% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-go-pgx/master/main.go %}
```

**Review by ianjevans — APPROVED** [review]
Looks good, just some tagging nits and the `remote_include` URL needs to be fixed.

**Review by ericharmeling — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10960#-:-Mfc5xat9HXtNHORC8UU:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans)

___
*[v21.1/build-a-go-app-with-cockroachdb.md, line 46 at r1](https://reviewable.io/reviews/cockroachdb/docs/10960#-MfEL3BMR28HCaBw43TO-r1-46:-MfZGCIl0kI5no9FGelo:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/23fad729e7a076dd88e8bd44509cba0e458373a6/v21.1/build-a-go-app-with-cockroachdb.md#L46)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
    {% include_cached copy-clipboard.html %}
```
</blockquote></details>

Done.
___
*[v21.1/build-a-go-app-with-cockroachdb.md, line 70 at r1](https://reviewable.io/reviews/cockroachdb/docs/10960#-MfEL3BMR28HCaBw43TO-r1-70:-MfZGBznE2gZ1cA78_Kn:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/23fad729e7a076dd88e8bd44509cba0e458373a6/v21.1/build-a-go-app-with-cockroachdb.md#L70)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
{% include_cached copy-clipboard.html %}
```
</blockquote></details>

Done.
___
*[v21.1/build-a-go-app-with-cockroachdb.md, line 94 at r1](https://reviewable.io/reviews/cockroachdb/docs/10960#-MfEL3BMR28HCaBw43TO-r1-94:-MfZG81u1Ht-PPuzcXmB:byof0eg) ([raw file](https://github.com/cockroachdb/docs/blob/23fad729e7a076dd88e8bd44509cba0e458373a6/v21.1/build-a-go-app-with-cockroachdb.md#L94)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
{% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-go-pgx/master/main.go %}
```
</blockquote></details>

Good catch!

Done.
___
*[v21.1/hello-world-go-pgx.md, line 19 at r1](https://reviewable.io/reviews/cockroachdb/docs/10960#-MfEL3BMR28HCaBw43TQ-r1-19:-MfZGKBjBeGxKket0-Mo:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/23fad729e7a076dd88e8bd44509cba0e458373a6/v21.1/hello-world-go-pgx.md#L19)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
{% include_cached copy-clipboard.html %}
```
</blockquote></details>

Done.

**Pull request merged**